systemd template and instructions (#508)
Impostor/Impostor: docs/Server-configuration.md

IMPOSTOR_AntiCheat__Enabled=true
IMPOSTOR_AntiCheat__BanIpFromGame=true
```

# systemd

Linux users will likely want to create a systemd service definition to manage running the Impostor application.  Benefits to using a systemd service include the following:

- The server process can automatically restart if the process dies
- Impostor can be set to start on boot (handy if you do system updates regularly)
- You can use systemd targets to ensure the process doesn't start until the system dependencies are met

Using your favorite text editor, create the following service file:

```

> sudo vim /etc/systemd/system/impostor.service

Reviewer: ooh, getting people stuck in vim, a true classic.

Maybe it's nicer to use `sudoedit` instead? Then it automatically opens in whatever editor they have configured

Author: that sounds reasonable to me

```

> Modify the following template to your needs and populate the service file.

Reviewer: Should we put the template in here or should we put it as a separate file that people can just copypaste? (or download from github and use directly)

```
[Unit]

> Description=Impostor private Among Us server - https://github.com/Impostor/Impostor

Reviewer: I'm looking at my own unit file (probably should've upstreamed that sooner but oops) and I also have two exta lines here:

```
After=syslog.target
After=network.target
```

This seems to be orthagonal to WantedBy, should I add these lines here or are they sort of obsolete?

Author: WantedBy=multiuser.target adds the multiuser target as a dependency.  I think it's a slightly cleaner way of doing it.  Ultimately your way would also likely work.

Reviewer: It's not `After` or `WantedBy`, you want both:
- After specifies this unit must start when a few other units have also started
- WantedBy gives a deadline: this unit must start for the system to consider itself to be fully started.

[Service]
# Directory where Impostor is installed
WorkingDirectory=/opt/impostor
# Path to Impostor binary
ExecStart=/opt/impostor/Impostor.Server
Restart=always
# Restart service after 10 seconds if the dotnet service crashes
RestartSec=10

> KillSignal=SIGINT

Reviewer: Hmm, I don't have this value specified in my config file. The default KillSignal seems to be SIGTERM, does SIGINT make a difference here? I tested it on my pc and in both cases the client was disconnected gracefully

Author: Let's switch it to SIGTERM.  SIGINT sends control + c and after doing some reading, SIGTERM makes more sense.

> SyslogIdentifier=log.impostor

Reviewer: Hmm, what is the effect of this line? Can you share an example log line?

For example, this is a log entry on my server:
```
Jan 19 23:15:24 euhtzfsn1 Impostor.Server[11330]: [23:15:24 WRN] Received RpcFlag for unregistered NetId 4.
```

Author: > Jan 20 02:25:39 amongus log.impostor[6741]: [02:25:39 INF] Client 3 disconnecting, reason: The remote sent a disconnect request

Reviewer: ah okay, it just changes the name then. Not sure why we'd need the `log.` prefix then, maybe remove that prefix?

> # User and group that will run the Impostor server process -- ensure that user has execute on the binary

Reviewer: ```suggestion
# User and group that will run the Impostor server process -- ensure that user is allowed to execute the binary
```

I think this is more clear to the more casual linux user

Author: That sounds good to me

User=impostor
Group=impostor
TimeoutStopSec=10

[Install]
# Wait until most system services have started before starting Impostor
WantedBy=multi-user.target
```

Reload all systemd unit files to source in the new file you've created.

```
sudo systemctl daemon-reload
```

> Start the service and verify that it starts correctly.  Hint: Use the Q key to quit out of journalctl.

Reviewer: Hmm, the hint only applies to the second command. Suggestion: add a small explanation after the first command and move the hint to there

Author: will do
